src/odemis/acq/stream/_live.py

```diff
@@ -84,14 +88,30 @@ def _onActive(self, active):
         """ Called when the Stream is activated or deactivated by setting the
         is_active attribute
         """
-        # Make sure the stream is prepared before really activate it
         if active:
-            if not self._prepared:
-                logging.debug("Preparing stream before activating it as it wasn't prepared")
-                self._prep_future = self._prepare()
-                self._prep_future.add_done_callback(self._startAcquisition)
+            if self.inactive_on_next_image.value:
+                # Check if the stream itself has a dedicated get single frame method
+                if hasattr(self, "getSingleFrame"):  # TODO K.K. extra check if this method is callable
+                    single_frame_func = self.getSingleFrame
+                else:
```

Comment:
It would be nice if you could avoid this check. Maybe by using inheritance, you could avoid that?